test/uk/gov/hmrc/fileupload/virusscan/VirusScannerSpec.scala

```diff
@@ -17,102 +17,69 @@
 import scala.concurrent.Future
-import scala.util.{Failure, Success}
 
-class VirusScannerSpec extends UnitSpec with Matchers with ScalaFutures with TestApplicationComponents {
+class VirusScannerSpec
+  extends UnitSpec
+     with Matchers
+     with ScalaFutures
+     with TestApplicationComponents
+     with IntegrationPatience {
 
-  private def enumerator(): Enumerator[Array[Byte]] = {
-    val inputString = "a random string long enough to by divided into chunks"
-    Enumerator[Array[Byte]](inputString.grouped(10).map(_.getBytes).toList: _*)
-  }
+  implicit val sys = ActorSystem("test")
+  implicit val mat = ActorMaterializer
 
-  val sendingChunksSuccessful = (_ : Array[Byte]) => Future.successful(())
-  val noVirusFound = () => Future.successful(Success(true))
-  val virusDetected = () => Future.successful(Failure(new VirusDetectedException("test virus")))
-  val commandTimeout = () => Future.successful(Failure(new VirusDetectedException("COMMAND READ TIMED OUT")))
+  val inputString = "a random string long enough to by divided into chunks"
+  private def fileSource: InputStream = new ByteArrayInputStream(inputString.getBytes)
+  val fileLength = inputString.getBytes.length
 
-  val virusScanner = new VirusScanner(ClamAntiVirus(_), components.configuration, components.environment)
+  def clean(inputstream: InputStream, length: Int) = Future.successful(Clean)
+  def virusDetected(inputstream: InputStream, length: Int) = Future.successful(Infected("test virus"))
+  def commandTimeout(inputstream: InputStream, length: Int) = Future.successful(Infected("COMMAND READ TIMED OUT"))
+  def failWith(exception: Exception)(inputStream: InputStream, length: Int): Future[ScanningResult] = Future.failed(exception)
 
-  "VirusScanner" should {
+  val virusScanner = new VirusScanner(AvClient(components.configuration, components.environment))
 
+  "VirusScanner" should {
     s"return $ScanResultFileClean result if input did not contain a virus" in {
-      val chunkEnumerator = enumerator()
-
-      val result = await(chunkEnumerator.run(new VirusScanner(
-        ClamAntiVirus(_), components.configuration, components.environment).scanIteratee(sendingChunksSuccessful, noVirusFound)).flatMap(identity))
-
+      val result = virusScanner.scanWith(sendAndCheck = clean)(fileSource, fileLength).futureValue
       result shouldBe Xor.Right(ScanResultFileClean)
     }
 
     s"return $ScanResultVirusDetected result if input contained a virus" in {
-      val chunkEnumerator = enumerator()
-
-      val result = await(chunkEnumerator.run(virusScanner.scanIteratee(sendingChunksSuccessful, virusDetected)).flatMap(identity))
-
-      result shouldBe Xor.Left(ScanResultVirusDetected)
+      val result = virusScanner.scanWith(sendAndCheck = virusDetected)(fileSource, fileLength).futureValue
+      result shouldBe Xor.left(ScanResultVirusDetected)
     }
 
     //clamav client unfortunately returns this as a virus detected exception
```

Comment:
is this comment still true?

Reply:
I believe the only difference is that the message is returned in an `Infected` result, rather than a VirusDetectedException - I'll update the comment